JavaScript/JavaScriptSDK/Init.ts

```diff
@@ -25,6 +25,27 @@ function initializeAppInsights() {
             init.emptyQueue();
 
             init.pollInteralLogs(appInsightsLocal);
+
+            // Add callback to push events when the user navigates away
+            // Note: This approach tries to push an async request with all the pending events onbeforeunload.
+            //       Firefox does not respect this. Other browsers DO push out the call with < 100% hit rate.
+            //       Telemetry here will help us analyze how effective this approach is.
+            //       Another approach would be to make this call sync with a acceptable timeout to reduce the 
+            //       impact on user experience.
+            if ('onbeforeunload' in window) {                
+                // Callback to flush all events
+                var flushAllEvents = function() {
+                    appInsightsLocal.trackEvent('AI (Internal): Flushing all events onbeforeunload');
+                    appInsightsLocal.context._sender.triggerSend();
+                };
+                
+                // check if addEventListener is available
+                if (window.addEventListener) {
```

Comment:
if (addEventListener)... else attachEvent is such a fundamental event triggering functionality, that I think we want to make it a utility function.